File: projects/pinstripe/packages/@pinstripe/core/lib/commands/list_commands.js

import chalk from 'chalk';

import { Command } from '../command.js';

Command.register('list-commands').define(dsl => dsl.props({
    run(){
        console.log('');
        console.log('The following commands are available:');
        console.log('');
        Object.keys(Command.classes).sort().forEach(commandName => {
            console.log(`  * ${chalk.green(commandName)}`);
        });
        console.log('');
    }
}));

File: projects/slick/lib/slick/commands/list_commands.rb

require "slick/command"

Slick::Command.register("list-commands").define_method "run" do

    puts ""
    puts "The following commands are available:"
    puts ""
    Slick::Command.registered_classes.keys.sort.each do |name|
        puts "  * #{name.green}"
    end
    puts ""

end


